src/env.rs

```diff
@@ -0,0 +1,123 @@
+use std::collections::HashMap;
+use std::env;
+use std::fmt;
+
+use heck::ShoutySnekCase;
+
+/// RoverEnv allows us to mock environment variables while
+/// running tests. That way we can run our tests in parallel,
+/// and our local development environment will not have unintended
+/// side effects on our tests.
+#[derive(Debug, Clone)]
+pub struct RoverEnv {
+    mock_store: Option<HashMap<String, String>>,
+}
+
+impl Default for RoverEnv {
+    fn default() -> RoverEnv {
+        RoverEnv::new()
+    }
+}
+
+impl RoverEnv {
+    /// creates a new environment variable store
+    pub fn new() -> RoverEnv {
+        let mock_store = if cfg!(test) {
+            Some(HashMap::new())
+        } else {
+            None
+        };
+
+        RoverEnv { mock_store }
+    }
+
+    /// returns the value of the environment variable if it exists
+    pub fn get(&self, key: RoverEnvKey) -> std::io::Result<Option<String>> {
+        let key = key.to_string();
+        match &self.mock_store {
+            Some(mock_store) => Ok(mock_store.get(&key).map(|v| v.to_owned())),
+            None => match env::var(&key) {
+                Ok(data) => Ok(Some(data)),
+                Err(e) => match e {
+                    env::VarError::NotPresent => Ok(None),
+                    env::VarError::NotUnicode(_) => Err(std::io::Error::new(
+                        std::io::ErrorKind::InvalidInput,
+                        format!(
+                            "The value of the environment variable \"{}\" is not valid Unicode.",
+                            &key
+                        ),
+                    )),
+                },
+            },
+        }
+    }
+
+    /// sets an environment variable to a value
+    pub fn set(&mut self, key: RoverEnvKey, value: &str) {
+        let key = key.to_string();
+        match &mut self.mock_store {
+            Some(mock_store) => {
+                mock_store.insert(key, value.into());
+            }
+            None => {
+                env::set_var(&key, value);
+            }
+        }
+    }
+
+    /// unsets an environment variable
+    pub fn unset_env(&mut self, key: RoverEnvKey) {
```

Comment:
Similar comment as for `set`/`insert`, I think this might also benefit from being renamed to `remove` like the [hashmap's `remove` API](https://doc.rust-lang.org/stable/std/collections/struct.HashMap.html#method.remove)